### ISSUE TITLE
Gmagick does not like tempnam without file extension

### ISSUE THREAD

**Issue by leofeyer** [contao/image#67]
The resizer uses a temporary file for atomic write operations:

https://github.com/contao/image/blob/9cd4150e52321f32aad1702128bca4c74bde2250/src/Resizer.php#L123-L127

This temporary file, however, does not have a file extension, which leads to an `No encode delegate for this image format (webp)` error in Gmagick on Ubuntu 16.04. The error did not occur in my dev environment on macOS.

Appending the file extension to the temporary file fixed the issue:

```php
// Atomic write operation
$tmpPath = $this->filesystem->tempnam($dir, 'img');
$this->filesystem->chmod($tmpPath, 0666, umask());

// Append the file extension to the temporary file
rename($tmpPath, $tmpPath.'.'.$imagineOptions['format']);
$tmpPath .= '.'.$imagineOptions['format'];

$imagineImage->save($tmpPath, $imagineOptions);
$this->filesystem->rename($tmpPath, $path, true);
```

Not sure if there is a better way to accomplish this though.

**Comment by ausi** [comment]
This should’ve already been fixed in 55a770a5ddd07fc5a5395fe46a8075a731c4d759 I think. We now always set the `$imagineOptions['format']` to make sure that the file extension has no effect.

The `No encode delegate` message from GMagick seems strange to me in this case. 😕

**Comment by leofeyer** [comment]
55a770a does **not** fix the issue for me. If the temporary file does not have the `.webp` extension, the error occurs. 🤷‍♂

**Comment by ausi** [comment]
I think this is a bug in Imagine or GMagick, see https://github.com/avalanche123/Imagine/issues/732

**Comment by ausi** [comment]
Until avalanche123/Imagine#733 gets merged, we could fix it with the following code I think:

```php
try {
	$imagineImage->save($tmpPath, $imagineOptions);
} catch(\GmagickException $e) {
	if (stripos($e->getMessage(), 'no encode delegate') === false) {
		throw $e;
	}
	
	$gmagick = new \Gmagick();
	$gmagick->newimage(1, 1, (new \GmagickPixel('#FFFFFF'))->getcolor(false));
	$gmagick->setimageformat(strtoupper($imagineOptions['format']));
	$gmagick->getImageBlob();
	
	$imagineImage->save($tmpPath, $imagineOptions);
}
```

**Comment by ausi** [comment]
@leofeyer should I implement the bugfix from https://github.com/contao/image/issues/67#issuecomment-600629586 ?

**Comment by leofeyer** [comment]
Yes please. 👍

**Comment by ausi** [comment]
See #70

I think we should keep this issue open though because we probably want to revert the fix once this issue is resolved in Imagine or Gmagick.

**Comment by leofeyer** [comment]
Definitely. 👍

**Comment by ausi** [comment]
Note to myself: Revert f246a592bb076dd544474708a90a96447dfaa280 once https://github.com/avalanche123/Imagine/pull/733 got resolved.

**Comment by ausi** [comment]
Closed in favor of #76
